src/_c_internal_utils.c

```diff
@@ -124,6 +124,47 @@ mpl_SetForegroundWindow(PyObject* module, PyObject *arg)
 #endif
 }
 
+static PyObject*
+mpl_SetDpiAwareness(PyObject* module)
+{
+#ifdef _WIN32
+#ifdef _DPI_AWARENESS_CONTEXTS_
+    // These functions and options were added in later Windows 10 updates, so
+    // must be loaded dynamically.
+    typedef BOOL (WINAPI *IsValidDpiAwarenessContext_t)(DPI_AWARENESS_CONTEXT);
+    typedef BOOL (WINAPI *SetProcessDpiAwarenessContext_t)(DPI_AWARENESS_CONTEXT);
+
+    HMODULE user32 = LoadLibrary("user32.dll");
+    IsValidDpiAwarenessContext_t IsValidDpiAwarenessContextPtr =
+        (IsValidDpiAwarenessContext_t)GetProcAddress(
+            user32, "IsValidDpiAwarenessContext");
+    SetProcessDpiAwarenessContext_t SetProcessDpiAwarenessContextPtr =
+        (SetProcessDpiAwarenessContext_t)GetProcAddress(
+            user32, "SetProcessDpiAwarenessContext");
+    if (IsValidDpiAwarenessContextPtr != NULL && SetProcessDpiAwarenessContextPtr != NULL) {
+        if (IsValidDpiAwarenessContextPtr(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2)) {
+            // Added in Creators Update of Windows 10.
+            SetProcessDpiAwarenessContextPtr(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE_V2);
+        } else if (IsValidDpiAwarenessContextPtr(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE)) {
+            // Added in Windows 10.
+            SetProcessDpiAwarenessContextPtr(DPI_AWARENESS_CONTEXT_PER_MONITOR_AWARE);
+        } else if (IsValidDpiAwarenessContextPtr(DPI_AWARENESS_CONTEXT_SYSTEM_AWARE)) {
+            // Added in Windows 10.
+            SetProcessDpiAwarenessContextPtr(DPI_AWARENESS_CONTEXT_SYSTEM_AWARE);
+        }
+        FreeLibrary(user32);
```

Comment:
this could be factored out of the if()

Reply:
Then it would be outside the `#if`.

Reply:
Ugh indeed.  Perhaps the preprocessor directives could avoid introducing braces, even if that means a tiny bit more repetition?

Reply:
Sure, the duplication doesn't really look that bad in the end, actually.